targets/nintendoswitch.s

```diff
@@ -71,7 +69,27 @@ bssloop:
     b bssloop
 
 run:
+    // process .dynamic section
+    adrp x0, _aslr_base
+    ldr  x0, [x0, #:lo12:_aslr_base]
+    adrp x1, _DYNAMIC
+    add  x1, x1, #:lo12:_DYNAMIC
+    bl   __dynamic_loader
+
+    // set LR to svcExitProcess if it's null
+    adrp x3, exit
+    add x3, x3, #:lo12:exit
+    cmp x30, xzr
+    csel x30, x3, x30, eq
+
     // call entrypoint
-    adrp x30, exit
-    add  x30, x30, #:lo12:exit
+    mov x3, sp
+    sub sp, sp, 0x10
+    stp x29, x30, [sp]
```

Comment:
Why do you need to store these two registers on the stack? Are they used by `svcExitProcess`?

It would make more sense to me to assume `main` never returns and exit it from within the Go `main` function, Go is much easier to understand than assembly.

Reply:
Oh, these are junk from older code. I will remove it.